assignment+4/Record.java

```diff
@@ -1,60 +1,99 @@
 // a custom class that represents each unique record from the GNIS file
-package assignment4;
-
-public class Record {
-  public String cgndbId;
-  public String geographicName;
-  public String genericTerm;
-  public float latitude;
-  public float longitude;
-  public String location;
-
-  // constructor for Record object
-  public Record(String l) {
-    // regex allows us to split only on commas that are outside quotes
-    String[] elements = l.split(",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
-    int size = elements.length;
-
-    this.cgndbId = elements[0];
-    this.geographicName = elements[1];
-    this.genericTerm = elements[2];
-    this.latitude = Float.parseFloat(elements[3]);
-    this.longitude = Float.parseFloat(elements[4]);
-    this.location = null;
-
-    // if there is a location, assign it to the class variable
-    if (size == 6) {
-      this.location = elements[5];
-    }
+
+public class Record implements Comparable<Record> {
+  private String cgndbId;
+  private String geographicName;
+  private String genericTerm;
+  private Double latitude;
+  private Double longitude;
+  private String location;
+  private String province;
+
+  public static String SORT_BY = "CGNDBID";
```

Comment:
What does this do?

Reply:
that public static variable (accessible anywhere in the program) will be used a switch to run comparisons in the Record class, in the compareTo() method in the Record class